homeassistant/components/climate/fritzhome.py

```diff
@@ -0,0 +1,137 @@
+"""
+Support for AVM Fritz!Box fritzhome thermostate devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/climate.fritzhome/
+"""
+import logging
+
+from homeassistant.components.fritzhome import DOMAIN
+from homeassistant.components.climate import (
+    ATTR_OPERATION_MODE, ClimateDevice, STATE_ECO,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import (PRECISION_HALVES)
+from homeassistant.const import (TEMP_CELSIUS, ATTR_TEMPERATURE)
+
+DEPENDENCIES = ['fritzhome']
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_COMFORT = 'comfort'
+STATE_MANUAL = 'manual'
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE)
+
+OPERATION_LIST = [STATE_COMFORT, STATE_ECO, STATE_MANUAL]
+
+MIN_TEMPERATURE = 8
+MAX_TEMPERATURE = 28
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Fritzhome thermostat platform."""
+    device_list = hass.data[DOMAIN]
+
+    devices = []
+    for device in device_list:
+        if device.has_thermostat:
+            devices.append(FritzhomeThermostat(hass, device))
+
+    add_devices(devices)
+
+
+class FritzhomeThermostat(ClimateDevice):
+    """The thermostat class for Fritzhome thermostates."""
+
+    def __init__(self, hass, device):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.